src/main.cpp

```diff
@@ -74,6 +81,54 @@ int main(int argc, char *argv[]){
             return 2;
         }
     }
+
+    // Generate the seed values:
+    char seq[100];
+    std::ifstream file("/dev/urandom", std::ios::in | std::ios::binary);
+    if(!file) {
+        std::cout << "Error opening file\n";
+    }
+    file.read(seq, sizeof(char)*100);
+    file.close();
+    
+    std::seed_seq seed(seq, seq+100);
+
+    std::mt19937_64 gen(seed);
+
+    std::uniform_int_distribution<int64_t> dist(0, 100000);
+    
+    // Set of Characters
+    std::string CharSet = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789";
+    // Set of symbol characters
+    std::string symblSet = "!@#$%^&*-_";
+
     
+    if(!symbol_set && !askd_VorH){
+        // Instansiate the RadomEngine class: for various cofigurations.
+        // 1. Instansiate the class RandomEngine for -L option.
+        RandomEngine el(length);
+        el.setSymbolSet(symbol_set);
+
+        // set the length of the password:
+        el.setLength(length);
+        
+        // the porpose of this loop is to generate random indexs in a vector.
+        // then use those random indexes to find the corresponding character in the Set string.
+        for(int i = 0; i < el.getLength(); i++)
+        {
+            int indx = dist(gen) % 62;
```

Comment:
Seems like you just copied the code in hurry. 
There are a lot of issues here.

* You don't need an array of indexes. You can generate an index and just choose a char from character set. You first generate index and then iterate over them which adds extra loops and drags down performance for no reason at all.
* `dist(gen)%62` why? just make a distribution with range `[0,62]` and you won't need to do mod. You just copied without understanding.